src/lib.rs

```diff
@@ -3561,6 +3563,23 @@ impl Connection {
         self.streams.has_readable() || self.dgram_recv_front_len().is_some()
     }
 
+    /// Returns true if the connection is draining.
+    ///
+    /// If this returns true, the connection object cannot yet be dropped, but
+    /// no new application data can be sent or received. An application should
```

Comment:
Can we describe what to expect if stream_send/dgram_send is called when `is_draining` is true?

I'm  not so worried about documenting receive, although someone not so familiar with QUIC might be confused about whether exisitng buffered data would be availalable when stream_recv() is called

Reply:
I think that would need actually deciding what we want to do in those cases. Right now both `stream_send()` and `stream_recv()` kind of ignore this state (for no real reason), but of course if you try sending stream data it's not actually sent on the wire.

Maybe we should return `Error::InvalidState` from `stream_send()` in this case? Though that's probably for a separate PR.

Reply:
Agree this change doesn't change what we already do, and what we already do or don't document. Happy to punt that to a separate follow up as long as we remember to get back to it.